src/ConnectedMode/Binding/BindingProcessImpl.cs

```diff
@@ -38,53 +38,24 @@ namespace SonarLint.VisualStudio.ConnectedMode.Binding
     internal class BindingProcessImpl : IBindingProcess
     {
         private readonly BindCommandArgs bindingArgs;
-        private readonly ISolutionBindingOperation solutionBindingOperation;
-        private readonly IBindingConfigProvider bindingConfigProvider;
-        private readonly ISonarQubeService sonarQubeService;
-        private readonly IConfigurationPersister configurationPersister;
         private readonly IExclusionSettingsStorage exclusionSettingsStorage;
-        private readonly IEnumerable<Language> languagesToBind;
+        private readonly ISonarQubeService sonarQubeService;
+        private readonly IQualityProfileDownloader qualityProfileUpdater;
         private readonly ILogger logger;
 
         public BindingProcessImpl(
             BindCommandArgs bindingArgs,
-            ISolutionBindingOperation solutionBindingOperation,
-            IBindingConfigProvider bindingConfigProvider,
             IExclusionSettingsStorage exclusionSettingsStorage,
-            IConfigurationPersister configurationPersister,
             ISonarQubeService sonarQubeService,
+            IQualityProfileDownloader qualityProfileUpdater,
             ILogger logger,
             bool isFirstBinding = false)
-            : this(bindingArgs,
-                  solutionBindingOperation,
-                  bindingConfigProvider,
-                  exclusionSettingsStorage,
-                  configurationPersister,
-                  sonarQubeService,
-                  logger,
-                  isFirstBinding,
-                  languagesToBind: Language.KnownLanguages)
-        { }
-
-        internal /* for testing */ BindingProcessImpl(
-            BindCommandArgs bindingArgs,
-            ISolutionBindingOperation solutionBindingOperation,
-            IBindingConfigProvider bindingConfigProvider,
-            IExclusionSettingsStorage exclusionSettingsStorage,
-            IConfigurationPersister configurationPersister,
-            ISonarQubeService sonarQubeService,
-            ILogger logger,
-            bool isFirstBinding,
-            IEnumerable<Language> languagesToBind)
         {
-            this.bindingArgs = bindingArgs ?? throw new ArgumentNullException(nameof(bindingArgs));
```

Comment:
I deleted the null checks because I mistakenly thought this class was MEF-created, in which case the arguments won't be null (and we don't now explicitly do null checks for MEF-created components).
However, it isn't created by MEF so I'll reinstate them.